### PR TITLE
prep release: v1.4.0

### PR DESCRIPTION
**Pull request by abernix**
This prepares for the v1.4.0 by:

- Bump versions to v1.4.0
- Update license file
- Straight copy of `NEXT_CHANGELOG.md` to `CHANGELOG.md`
- CHANGELOG.md editorial prior to 1.4.0 release.

### CODE REVIEW

File: docs/source/containerization/kubernetes.mdx [hunk 1]
```diff
@@ -64,10 +64,10 @@ kind: ServiceAccount
 metadata:
   name: release-name-router
   labels:
-    helm.sh/chart: router-1.0.0-rc.7
+    helm.sh/chart: router-1.0.0-rc.8
```
Comment (garypen) [inline docs/source/containerization/kubernetes.mdx:67]: Is it time to make this 1.0? We have had at least one request for this to happen and it's been stable for a while now.

File: helm/chart/router/README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 [router](https://github.com/apollographql/router) Rust Graph Routing runtime for Apollo Federation
 
-![Version: 1.0.0-rc.7](https://img.shields.io/badge/Version-1.0.0--rc.7-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: v1.3.0](https://img.shields.io/badge/AppVersion-v1.3.0-informational?style=flat-square)
+![Version: 1.0.0-rc.7](https://img.shields.io/badge/Version-1.0.0--rc.7-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: v1.4.0](https://img.shields.io/badge/AppVersion-v1.4.0-informational?style=flat-square)
```
Comment (garypen) [inline helm/chart/router/README.md:5]: Was this generated using `helm-docs`? It should be at `rc.8`...

**Review by bnjjj — APPROVED** [review]

File: helm/chart/router/README.md (hunk 2 above)
Comment (abernix) [reply helm/chart/router/README.md]: Eh, that's me.  It was, and then I had to make another change.  Fixed in a4b73f5.

File: docs/source/containerization/kubernetes.mdx (hunk 1 above)
Comment (abernix) [reply docs/source/containerization/kubernetes.mdx:67]: Happy to entertain this literally tomorrow.  Feel free to open a PR!

**Review by garypen — APPROVED** [review]

**Review by o0Ignition0o — APPROVED** [review]
One suggestion, otherwise lgtm

**Review by Geal — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,113 @@ All notable changes to Router will be documented in this file.
...
+
+### **Experimental** 🥼 External cache storage in Redis ([PR #2024](https://github.com/apollographql/router/pull/2024))
+
+We are experimenting with introducing external storage for caches in the Router, which will provide a foundation for caching things like automated persisted queries (APQ) amongst other future-looking ideas.  Our initial implementation supports a multi-level cache hierarchy, first attempting an in-memory LRU-cache, proceeded by a Redis Cluster backend.
+
+As this is still experimental, it is only available as an opt-in through a Cargo feature-flag.
+
+By [@garypen](https://github.com/garypen) and [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/2024
+
+### Expose `query_plan` to `ExecutionRequest` ([PR #2081](https://github.com/apollographql/router/pull/2081))
```
[the remark this answers is not in the document]
Comment (abernix) [reply CHANGELOG.md]: Accepted, but changed it to be Rhai rather than RHAI to match the format they use in their project.

**Pull request merged**